src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -197,6 +198,11 @@ internal DecisionTree ComputeDecisionTree()
                         else
                         {
                             AddToDecisionTree(result, label);
+                            if (label.Guard == null || label.Guard.ConstantValue == ConstantValue.True)
+                            {
+                                // Only unconditional switch labels contribute to subsumption
+                                AddToDecisionTree(subsumptionTree, label);
```

Comment:
No, `CheckSubsumed` only checks whether the a pattern is subsumed by a decision tree. It doesn't modify the tree.